src/components/RegisterForm/index.js

```diff
@@ -0,0 +1,383 @@
+      nameValid = false;
+      $firstNameErrorMessage.classList.add('show-error');
+      $firstNameErrorMessage.innerText = 'Informe seu nome';
+      firstNameInput.inputError();
+    }
+
+    if (!this.isSurnameValid(surnameValue)) {
+      surnameValid = false;
+      $surnameErrorMessage.classList.add('show-error');
+      $surnameErrorMessage.innerText = 'Informe seu sobrenome';
+      surnameInput.inputError();
+    }
+
+    if (!this.isBirthdayValid(birthdayValue)) {
+      birthdayValid = false;
+      $birthdayErrorMessage.classList.add('show-error');
+      $birthdayErrorMessage.innerText = 'Informe sua data de nascimento';
+      birthdayInput.inputError();
+    }
+
+    if (!dropdownValue.value) {
+      dropdownValid = false;
+      $dropdownErrorMessage.classList.add('show-error');
+      dropdownInput.selected
+        .get('dropdown-toggle')
+        .classList.add('dropdown-form__show-error-dropdown');
+      $dropdownErrorMessage.innerText = 'Selecione sua cidade';
+    }
+
+    if (!this.isEmailValid(emailValue)) {
+      emailValid = false;
+      $emailErrorMessage.classList.add('show-error');
+      $emailErrorMessage.innerText = 'Informe um e-mail válido';
+      emailInput.inputError();
+    }
+
+    if (!this.isPhoneValid(phoneValue)) {
+      phoneValid = false;
+      $phoneErrorMessage.classList.add('show-error');
+      $phoneErrorMessage.innerText = 'Informe um número de telefone válido';
+      phoneInput.inputError();
+    }
+
+    if (!this.isPasswordValid(passwordValue)) {
+      passwordValid = false;
+      $passwordErrorMessage.classList.add('show-error');
+      $passwordErrorMessage.innerText =
+        'Senha inválida. Sua senha deve conter no mínimo 10 caracteres, incluindo pelo menos um caractere especial e uma letra maiúscula.';
+      passwordInput.inputError();
+    }
+
+    if (
+      !this.isConfirmPasswordValid(confirmPasswordValue) ||
+      passwordValue !== confirmPasswordValue
+    ) {
+      confirmPasswordValid = false;
+      $confirmPasswordErrorMessage.classList.add('show-error');
+      $confirmPasswordErrorMessage.innerText = 'Senhas diferentes';
+      confirmPasswordInput.inputError();
+    }
+
+    if (nameValid) $firstNameErrorMessage.classList.remove('show-error');
+    if (surnameValid) $surnameErrorMessage.classList.remove('show-error');
+    if (birthdayValid) $birthdayErrorMessage.classList.remove('show-error');
+    if (dropdownValid) {
+      $dropdownErrorMessage.classList.remove('show-error');
+      dropdownInput.selected
+        .get('dropdown-toggle')
+        .classList.remove('dropdown-form__show-error-dropdown');
+    }
+    if (emailValid) $emailErrorMessage.classList.remove('show-error');
+    if (phoneValid) $phoneErrorMessage.classList.remove('show-error');
+    if (passwordValid) $passwordErrorMessage.classList.remove('show-error');
+    if (confirmPasswordValid)
+      $confirmPasswordErrorMessage.classList.remove('show-error');
```

Comment:
Aqui seria melhor usar o toggle do classList, assim n precisaria adicionar a classe lá em cima e depois remover ela aqui em baixo, estaria tudo junto em uma linha só de código.

Reply:
Fiz aqui uns testes trocando pelo toggle e nesse caso ele teve um comportamento estranho. Quando a validação for verdadeira ele deveria esconder a mensagem de erro, mas se o input já tiver com erro ativado msm que vc preencha corretamente os campos e tente apertar de novo em cadastrar ele não some com a mensagem de erro (imagem abaixo). Fazendo pelo .add e .remove como já tava não aconteceu isso 🤔

![image](https://github.com/devhatt/pet-dex-frontend/assets/145572554/9b432936-c12d-4a88-ba52-6dc0cee4e0f9)

Reply:
Entendi, deixa do seu jeito mesmo então